crypto/cms/cms_sd.c

```diff
@@ -745,8 +749,13 @@ int CMS_SignerInfo_verify(CMS_SignerInfo *si)
         return -1;
     }
     mctx = si->mctx;
+    if (si->pctx != NULL) {
+        EVP_PKEY_CTX_free(si->pctx);
+        si->pctx = NULL;
+    }
```

Comment:
While working out a PR for master, I realised that this function
can be called twice if e.g. CMS_verify is called twice, and that would
create a memory leak.